qemu/tests/blockdev_mirror_cancel_ready_job_with_ioerror.py

```diff
@@ -0,0 +1,116 @@
+from provider.blockdev_mirror_nowait import BlockdevMirrorNowaitTest
+from provider.job_utils import get_event_by_condition
+
+
+class BlockdevMirrorCancelReadyIOError(BlockdevMirrorNowaitTest):
+    """
+    Cancel a ready job when target image is in error
+    """
+
+    def __init__(self, test, params, env):
+        params['filter-node-name'] = params['filter_node_name']
+        super(BlockdevMirrorCancelReadyIOError, self).__init__(test, params, env)
+
+    def _blockdev_add_images(self, images):
+        for tag in images:
+            params = self.params.object_params(tag)
+            devices = self.main_vm.devices.images_define_by_params(tag,
+                                                                   params,
+                                                                   'disk')
+            devices.pop()
+            for dev in devices:
+                if self.main_vm.devices.get_by_qid(dev.get_qid()):
+                    continue
+                ret = self.main_vm.devices.simple_hotplug(dev,
+                                                          self.main_vm.monitor)
+                if not ret[1]:
+                    self.test.fail("Failed to hotplug '%s': %s."
+                                   % (dev, ret[0]))
+
+    def _create_images(self, images):
+        for tag in images:
+            disk = self.source_disk_define_by_params(self.params, tag)
+            disk.create(self.params)
+            self.trash.append(disk)
+
+    def create_source_images(self):
+        """create source images of data images"""
+        self._create_images(self.params.objects("source_images"))
+
+    def create_target_images(self):
+        """create target images of mirror images"""
+        self._create_images(self.params.objects("target_images"))
+
+    def add_source_images(self):
+        """blockdev-add source images: protocol and format nodes only"""
+        self.create_source_images()
+        self._blockdev_add_images(self.params.objects("source_images"))
```

Comment:
hi @aliang123 , I suggest we use a single source and target image here, instead of a list as we did currently.
Note from line49 to line 83, we only handle one source and one target and one mirror job, but here we tried to add multiple images(line15 and line31). IMO, one single image is OK for the test, what's your opinion?